pkg/scheduler/framework/interface.go

```diff
@@ -608,6 +613,36 @@ type Handle interface {
 	Parallelizer() parallelize.Parallelizer
 }
 
+// PreFilterResult wraps needed info for scheduler framework to act upon PreFilter phase.
+type PreFilterResult struct {
+	// The set of nodes that should be considered downstream; if nil then
+	// all nodes are eligible.
+	NodeNames sets.String
+}
+
+func (p *PreFilterResult) Merge(in *PreFilterResult) *PreFilterResult {
+	if p == nil && in == nil {
```

Comment:
If we can invoke `merge`, p can't be nil, right?

Reply:
it can in golang, only accessing variables will cause panic

Reply:
Yeah. You are right.  I forgot that a struct of nil can call a method without causing a panic